service/clients/oracle/metrics_client.go

```diff
@@ -1,33 +1,35 @@
-package client
+package oracle
 
 import (
 	"context"
 	"time"
 
 	"cosmossdk.io/log"
+	"google.golang.org/grpc"
 
-	"github.com/skip-mev/slinky/service"
 	"github.com/skip-mev/slinky/service/metrics"
+	"github.com/skip-mev/slinky/service/servers/oracle/types"
 )
 
-var _ service.OracleService = (*MetricsClient)(nil)
+var _ OracleClient = (*MetricsClient)(nil)
```

Comment:
I feel like since the only difference between a metrics client and the normal grpc client is that the metrics are enabled, they should basically just be the same struct with the same routine (with the extra metrics stuff added if enabled.  Its mainly duplicate code

Reply:
yea agreed, will update

Reply:
at second glance this is probably better deferred to a different task, https://linear.app/skip/issue/BLO-802/allow-the-oracle-client-to-continue-to-restart-in-the-case-it-cannot. We are going to want to build in some type of restart logic into the client in case it ever loses its connection.

Reply:
Agree w/ Alex. Maybe using a metrics interface is the easier solution here instead of embedding an OracleClient? That way we can just construct w/ a no-op if not enabled and have exactly the same methods/code across both situations.